### ISSUE TITLE
Size limit of User Data Folder

### ISSUE THREAD

**Issue by exaiwitmx** [MicrosoftEdge/WebView2Feedback#3243]
Is there a size limit for the User Data Folder?
If so, what is the default size limit? 
Is there any means for an application to set this limit? In my case, it's a WinForms application.
Is the User Data Folder purged automatically if the limit is reached?

**Comment by novac42** [comment]
I don't think there's default size limit for User Data folder in WebView2 but adding @LiangTheDev to confirm. 
@exaiwitmx Would you like to provide more background for the scenario that may generate a large amount of user data?

**Comment by exaiwitmx** [comment]
@novac42 

> @exaiwitmx Would you like to provide more background for the scenario that may generate a large amount of user data?
>
A customer of us has asked us this question. I do not know if it does actually happen in reality. 
We have set a custom User Data Folder location (inside %applocaldata%). 
There are customers which restrict the maximum size of the user profile, and they do not want the browser to eat up too much of the allowed space. 
Apart from that, if WebView2 has no size limit and no auto-purge, it seems very possible that the size of the UDF could grow more and more over time.

**Comment by LiangTheDev** [comment]
I don't think that there is a global limit over user data folder size for WebView2. However, most parts in the folder should not grow forever.
For disk caches (like http and code cache) that could grow, Chromium code has some heuristics on the default max size that takes account of available disk space. When the max size target is reached, trimming will happen.
If the app does not visit a lot of sites or having a service worker caching a lot of resources, the default behavior should be OK.
There is a [--disk-cache-size](https://peter.sh/experiments/chromium-command-line-switches/#disk-cache-size) Chromium command line switch that you could use to roughly control the max size of each disk cache.

There is currently one folder that could potentially grow out of control, that is the crash dumps folder. You could get the path of the folder using CoreWebView2Environment.FailureReportFolderPath. If there is a lot of WebView2 browser process crashes, there would be a lot of .dmp files created under that folder. We might limit the number of dmp files in the folder in the future, but before we do that, there is currently a risk that this folder could grow very large and you might have to delete .dmp files from the folder if it has grown too large.
